### PR TITLE
Ho 40 individual user sign up page back end

### PR DESCRIPTION
**Pull request by Sabirah42**
We've completed the backend and all tests are passing. 

We've not refactored with Ryan's authentication code as yet as we want to build the whole route and get it working first, so it'll be easier to change and debug later.

### CODE REVIEW

**Review by nadiangelica — APPROVED** [review]

File: backend/controllers/indUsersController.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const IndUser = require('../models/indUserModel');
+const mongoose = require('mongoose');
+
+//get all individual users
+const getAllIndUsers = async (req, res) => {
```
Comment (tlchambers) [inline backend/controllers/indUsersController.js:5]: There wouldn't be a case where we would need to getAllIndUsers, but we would need a getIndUserByID

File: backend/controllers/indUsersController.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const indUsers = await IndUser.find({}).sort({createdAt: -1})
+    res.status(200).json(indUsers)
+}
+
+//create a new individual user
+const createIndUser = async (req,  res) => {
+    const {firstName, surname, email, password} = req.body
```
Comment (tlchambers) [inline backend/controllers/indUsersController.js:12]: Here if we use firstName then 'surname' should be 'last name' so: 
`{firstName, lastName, email, password}`
or
`{forename, surname, email, password}`

It's a english language rule that I was told but I don't remember the exact reason why

File: backend/controllers/indUsersController.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    try {
+      const indUser = await IndUser.create({firstName, surname, email, password})
```
Comment (tlchambers) [inline backend/controllers/indUsersController.js:15]: again here too

File: backend/models/indUserModel.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require('mongoose')
+
+const validateEmail = (email) => {
```
Comment (tlchambers) [inline backend/models/indUserModel.js:3]: So me and Ryan looked at the validation rules and Ryan suggested we should use Mongoose validations (reference check out Ryan's branch) if you follow the OrgSchema, the codes are much cleaner :grimacing:

File: backend/routes/indUsers.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const express = require('express')
+const {
+    createIndUser,
+    getAllIndUsers
```
Comment (tlchambers) [inline backend/routes/indUsers.js:4]: line 4: getIndUserByID

File: backend/server.js [hunk 6]
```diff
@@ -19,19 +20,20 @@ app.use((req, res, next) => {
...
+// // CORS
+// const cors = require('cors');
+// const corsOptions = {
+//   origin: 'http://localhost:3000',
+//   credentials: true,
+//   optionSuccessStatus: 200,
+// };
 
-app.use(cors(corsOptions));
+// app.use(cors(corsOptions));
```
Comment (tlchambers) [inline backend/server.js:31]: We can remove all CORS

File: backend/spec/controllers/indUsersController.spec.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        });
+      expect(response.statusCode).toBe(201);
+      expect(response.body).toEqual({
+        message: "Thanks! your account has been successfully created",
+      });
+    });
+  });
+
+    describe("POST, when user first name, surname, email or password are invalid", () => {
+    test("the response code is 400 and returns an error stating first name is required", async () => {
```
Comment (tlchambers) [inline backend/spec/controllers/indUsersController.spec.js:30]: indentation on line 30

**Review by tlchambers — APPROVED** [review]

File: backend/controllers/indUsersController.js (hunk 1 above)
Comment (Sabirah42) [reply backend/controllers/indUsersController.js:5]: Aah, I left a comment about this on Ryan's code - I thought we were keeping it so we could easily test on Postman?

Comment (mantaslv) [reply backend/controllers/indUsersController.js:5]: yeah i feel like its good to leave in, we can take out if/before we go live

File: backend/controllers/indUsersController.js (hunk 2 above)
Comment (Sabirah42) [reply backend/controllers/indUsersController.js:12]: I ended up merging backend to frontend locally yesterday so we could keep working, so I'll change this in HO-38 and amend that PR as as it's not been approved yet

Comment (Sabirah42) [reply backend/controllers/indUsersController.js:12]: Have gone with lastName, FYI

File: backend/models/indUserModel.js (hunk 4 above)
Comment (Sabirah42) [reply backend/models/indUserModel.js:3]: We'll take a look and refactor once Ryan's has been merged into main, as we'll be able to compare a little easier that way

File: backend/routes/indUsers.js (hunk 5 above)
Comment (Sabirah42) [reply backend/routes/indUsers.js:4]: We left out that function as we weren't sure we'd need it for any functionality for v2 - I've added a ticket so we can discuss on Mon

File: backend/spec/controllers/indUsersController.spec.js (hunk 7 above)
Comment (Sabirah42) [reply backend/spec/controllers/indUsersController.spec.js:30]: Good spot!

**Pull request merged**